pkg/types/error.go

```diff
@@ -0,0 +1,100 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+)
+
+type ErrorCode int
```

Comment:
Use string instead of int. We can be more verbose since we're not dealing with C.